cmd/saturn-l2/main.go

```diff
@@ -234,6 +236,7 @@ func main() {
 			if err := l1client.Start(nConnectedL1s); err != nil {
 				if !errors.Is(err, context.Canceled) {
 					log.Errorw("terminated connection attempts with l1", "l1", l1ip, "err", err)
+					fmt.Printf("ERROR: Saturn Node was not able to connect to the L1 with IP %s after %f attempts, giving up.\n", l1ip, l1client.MaxReconnectAttempts)
```

Comment:
Station events are for non-technical users, we should keep them simple and easy to understand.

As a non-technical Station user, I don't understand details about L1/L2 interop, the IP address of an L1 node would be just confusing to me.

Also it's not important to me whether my Station L2 node is not able to connect to a particular L1 node, as long as I am connected to the network (to other L1 nodes) and can earn my FILs.

IMO, when L2 is not able to connect to a particular L1 node, it should pick a different nearby L1 node and try to connect to that one instead. IMO, there is no need to surface this internal Saturn detail in the Station's Activity Log.

For now, I am proposing to simply remove this log line from the PR.

Reply:
Good catch, I didn't evaluate it from this perspective.

I agree that the interop is none of the user's concerns. It would still be useful during development to get this information however. Currently we don't have a way to forward the debug log in case of non-critical crashes, right? I wonder if this is another argument for supporting multiple log levels in the activity stream, or whether we could add a menu bar item for inspecting the full log somewhoe.

Reply:
> It would still be useful during development to get this information however. Currently we don't have a way to forward the debug log in case of non-critical crashes, right? 

Currently, Station forwards everything from L2 node's stdout and stderr to Station's stdout/stderr, respectively.

See https://github.com/filecoin-project/filecoin-station/blob/f41f69acd0935d3a651fcba8c3b62b4b20e26517/main/saturn-node.js#L96-L103

So as long as you start the Station from your terminal, you can see all L2 node logs at a glance.


> I wonder if this is another argument for supporting multiple log levels in the activity stream, or whether we could add a menu bar item for inspecting the full log somehow.

I like the idea to allow us to access the full Station Module log from Station UI. Could you please open a new GH issue on filecoin-station?

Reply:
cool, I think starting from terminal is good enough for now. Once we do user testing we might want to see this information too, without requiring the user to launch the app from the terminal. I'll open an issue for this 👍

Reply:
https://github.com/filecoin-project/filecoin-station/issues/94